src/middlewared/middlewared/plugins/directoryservices_/connection.py

```diff
@@ -183,15 +183,15 @@ def _test_is_joined(self, ds_type: DSType, domain: str) -> bool:
                 is_joined_fn = self._ipa_test_join
             case _:
                 raise CallError(
-                    f'{ds_type}: specified directory service type does not '
-                    'support domain join functionality.'
+                    f'{ds_type}: The configured directory service type does not support joining a domain.',
+                    errno.EOPNOTSUPP
                 )
 
         return is_joined_fn(ds_type, domain)
```

Comment:
`ds_type` is not used in the 'test_join' functions.  If param is removed, then...
```suggestion
        return is_joined_fn(domain)
```